### PR TITLE
VS-77: Add more enum test cases

### PR DESCRIPTION
**Pull request by Ravi-Raghavan**
Added more Enum Test Cases for LINQ covering situations where the Enum is in a variable and that variable is referenced in the query(i.e. either query syntax or the method invocation syntax).

### CODE REVIEW

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinqExpressions.cs [hunk 1]
```diff
@@ -130,5 +130,15 @@ public void Query_syntax()
                 where ReturnArgument(ReturnArgument(ReturnArgument(this).ReturnArgument(person.Vehicle.VehicleType))).Type == VehicleTypeEnum.Bus
                 select person;
         }
+
+        [InvalidLinq("Convert(Convert((Convert({document}{EnumInt16}, Int32) + 10), GenType_Enum_12), Int32) is not supported.")]
+        [InvalidLinq3("Cannot find serializer for Convert(u.EnumInt16, Int32).", version: "[2.14.0-beta1, 2.18.0)")]
```
Comment (BorisDog) [inline tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinqExpressions.cs:135]: Let's have constants for versions (DriverVersions)

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs [hunk 2]
```diff
@@ -101,5 +101,130 @@ public void Query_Syntax()
...
+                .Where(u => u.EnumInt32 == EnumInt32.Value0 + 25);
+
+            _ = GetMongoQueryable<EnumHolder>()
+                .Where(u => u.EnumInt32 == EnumInt32.Value0 - 25);
+
+            _ = GetMongoQueryable<EnumHolder>()
+                .Where(u => u.EnumInt64 == EnumInt64.Value0 + 35);
+
+            _ = GetMongoQueryable<EnumHolder>()
+                .Where(u => u.EnumInt64 == EnumInt64.Value0 - 35);
```
Comment (BorisDog) [inline tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs:227]: Lets add more test cases from [here](https://github.com/mongodb/mongo-csharp-driver/commit/02957993aef5188667e24359c6e64c961f6f7514#diff-f0f89ad96b11379f9e941bb575b7a896bb2a2e15fca0fe42ba6204e78f26ec4d).

**Review by BorisDog — CHANGES_REQUESTED** [review]

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinqExpressions.cs (hunk 1 above)
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/NotSupportedLinqExpressions.cs:135]: Done

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs [hunk 3]
```diff
@@ -101,5 +101,130 @@ public void Query_Syntax()
...
+
+            var firstSignedEnum = EnumInt64.Value0;
+            var secondSignedEnum = EnumInt64.Value999;
+            var thirdSignedEnum = EnumInt64.MaxValue;
+
+            var firstVehicleType = VehicleTypeEnum.Bus;
+            var secondVehicleType = VehicleTypeEnum.Motorcylce;
+
+            _ = GetMongoQueryable<EnumHolder>()
+                .Where(u => u.EnumUInt64 == firstUnsignedEnum || u.EnumUInt64 == secondUnsignedEnum || u.EnumUInt64 == thirdUnsignedEnum)
```
[the remark this answers is not in the document]
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs:154]: Done

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs [hunk 4]
```diff
@@ -101,5 +101,130 @@ public void Query_Syntax()
...
+        [MQL("aggregate([{ \"$match\" : { \"Vehicle.VehicleType.Type\" : vehicleType } }, { \"$project\" : { \"LicenceNumber\" : \"$Vehicle.LicenceNumber\", \"_id\" : vehicleType } }])")]
+        public void Enum_single_variable()
+        {
+            var vehicleType = VehicleTypeEnum.Bus;
+
+            _ = GetMongoQueryable<Person>()
+                .Where(u => u.Vehicle.VehicleType.Type == vehicleType)
+                .Select(u => u.Vehicle.LicenceNumber);
+
+            VehicleTypeEnum? nullibleVehicleType = VehicleTypeEnum.Bus;
```
[the remark this answers is not in the document]
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs:114]: Done

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs [hunk 5]
```diff
@@ -101,5 +101,130 @@ public void Query_Syntax()
...
+
+            _ = GetMongoQueryable<Person>()
+                .Where(u => u.Vehicle.VehicleType.Type == nullibleVehicleType)
+                .Select(u => u.Vehicle.LicenceNumber);
```
[the remark this answers is not in the document]
Comment (Ravi-Raghavan) [reply tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs:118]: Created another ticket for Nullable support(i.e. VS-93) will implement changes in that ticket

File: tests/MongoDB.Analyzer.Tests.Common/DriverVersions.cs [hunk 6]
```diff
@@ -21,7 +21,13 @@ public static class DriverVersions
         public const string Linq3NonDefault = "[2.14.0-beta1, 2.19.0)";
         public const string Linq3DefaultAndHigher = V2_19_AndHigher;
         public const string Linq2DefaultAndLower = V2_18_AndLower;
+
+        public const string V2_14_Beta1_Till_V2_18 = "[2.14.0-beta1, 2.18.0)";
```
Comment (BorisDog) [inline tests/MongoDB.Analyzer.Tests.Common/DriverVersions.cs:25]: Where is this used?

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs [hunk 7]
```diff
@@ -101,5 +101,186 @@ public void Query_Syntax()
...
+            _ = GetMongoQueryable<EnumHolder>()
+                .Where(u => u.EnumInt16 == EnumInt16.Value0 - 15);
+
+            _ = GetMongoQueryable<EnumHolder>()
+                .Where(u => u.EnumInt32 == EnumInt32.Value0 + 25);
+
+            _ = GetMongoQueryable<EnumHolder>()
+                .Where(u => u.EnumInt32 == EnumInt32.Value0 - 25);
+
+            _ = GetMongoQueryable<EnumHolder>()
```
Comment (BorisDog) [inline tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqEnums.cs:279]: What about left-hand operations: `.Where(u => u.EnumInt32 + 123 == EnumInt32.Value0 - 25)`

**Review by BorisDog — CHANGES_REQUESTED** [review]

**Pull request closed without merging**